packages/services/f-analytics/src/store/_tests/analytics.module.test.js

```diff
@@ -1,71 +1,361 @@
+    newEvent
 } from '../../tests/helpers/setup';
+import {
+    PUSH_PLATFORM_DATA,
+    PUSH_USER_DATA,
+    PUSH_EVENT,
+    CLEAR_EVENTS
+} from '../mutation-types';
 
 const { actions, mutations } = AnalyticsModule;
-const { updatePlatformData, updateUserData } = actions;
-const { pushPlatformData, pushUserData } = mutations;
-let state = AnalyticsModule.state();
+const { pushPlatformData, pushUserData, pushEvent } = actions;
+
+describe('Analytics Module ::', () => {
+    let state;
+    let commit;
+
+    beforeEach(() => {
+        // Arrange
+        state = Object.assign({}, defaultState); // eslint-disable-line
+        commit = jest.fn((mutation, mutationPayload) => {
+            // Replace the current state with the modified during the test
+            // thus allow assertions on the final state
+            if (mutation === PUSH_PLATFORM_DATA) {
+                state.platformData = { ...state.platformData, ...mutationPayload };
+            }
+            if (mutation === PUSH_USER_DATA) {
+                state.userData = mutationPayload;
+            }
+            if (mutation === PUSH_EVENT) {
+                state.events = [...state.events, mutationPayload];
+            }
+            if (mutation === CLEAR_EVENTS) {
+                state.events = [];
+            }
+        });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
 
-describe('AnalyticsModule', () => {
     it('should create default state when initialised.', () => {
         // Assert
-        expect(state).toEqual(defaultState);
+        expect(AnalyticsModule.state()).toEqual(defaultState);
     });
 
-    describe('mutations ::', () => {
-        beforeEach(() => {
-            // Arrange
-            state = defaultState;
+    describe('actions ::', () => {
+        describe('pushPlatformData ::', () => {
+            it('should call the `pushPlatformData` mutation', () => {
+                // Act
+                pushPlatformData({ commit, state }, modifieldState.platformData);
+
+                // Assert
+                expect(commit).toHaveBeenLastCalledWith('pushPlatformData', modifieldState.platformData);
+            });
+
+            it('should push the `platformData` if clientside & dataLayer present', () => {
+                // Arrange
+                const windowsPushSpy = jest.fn();
+                const originalWindow = { ...window };
+                jest.spyOn(global, 'window', 'get').mockImplementation(() => ({
+                    ...originalWindow,
+                    dataLayer: {
+                        push: windowsPushSpy
+                    }
+                }));
+
+                // Act
+                pushPlatformData({ commit, state }, modifieldState.platformData);
+
+                // Assert
+                expect(windowsPushSpy).toHaveBeenLastCalledWith({ platformData: { ...modifieldState.platformData } });
+                expect(state.platformData).toEqual(modifieldState.platformData);
```

Comment:
You have separate tests to check that mutations update the state, I would remove these checks from actions to keep tests focused on actions only